controllers/secret_controller.go

```diff
@@ -0,0 +1,232 @@
+package controllers
```

Comment:
This is hard to review. It is not obvious what changes you are making. Please make a commit where you are restructuring (`git mv`) and then another commit where you are making the changes needed for the SDK port.